sympy/solvers/inequalities.py

```diff
@@ -964,6 +964,9 @@ def reduce_inequalities(inequalities, symbols=[]):
     for i in inequalities:
         if isinstance(i, Relational):
             i = i.func(i.lhs.as_expr() - i.rhs.as_expr(), 0)
+        elif "&" in str(i) or "|" in str(i) or "!" in str(i):
+            raise NotImplementedError(
```

Comment:
No need to convert to string: use `if i.has(And, Or, Not):`

Reply:
This same argument also applies to the implementation of `reduce_boolean_expr`

Reply:
On using Or, Not flake 8 is failing. That is why I chose to convert to string.

Reply:
@smichr @sylee957

Reply:
Flake8 error is because of the missing imports, and should be fixed easily.